etcd/src/integration/scala/io/buoyant/etcd/EtcdIntegrationTest.scala

```diff
@@ -0,0 +1,243 @@
+package io.buoyant.etcd
+
+import com.twitter.conversions.time._
+import com.twitter.finagle.{Http, Path}
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.io.Buf
+import com.twitter.util.{Events => _, _}
+import io.buoyant.test.{Awaits, Events}
+import java.io.File
+import java.util.UUID
+import org.jboss.netty.handler.codec.http._
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.fixture.FunSuite
+import scala.collection.JavaConverters._
+import scala.sys.process.{Process, ProcessLogger}
+
+/**
+ * Etcd client integration tests.
+ *
+ * Boots an etcd instance on localhost:4001.
+ */
+class EtcdIntegrationTest extends FunSuite with Awaits with BeforeAndAfterAll {
+
+  val processBuilder = Process(Seq("etcd", "--data-dir", "/tmp/etcd"))
+  var process: Process = null
+
+  override def beforeAll {
+    try {
+      process = processBuilder.run(ProcessLogger(new File("/dev/null")))
+    } catch {
+      case e: Exception => throw new Exception(s"etcd failed to start: ${e.getMessage}")
+    }
+  }
+
+  override def afterAll {
+    if (process != null) {
+      process.destroy
+    }
+  }
+
+  def serverName =
+    Option(System.getProperty("ETCD_NAME")) match {
+      case None | Some("") => "/$/inet/127.1/4001"
+      case Some(name) => name
+    }
```

Comment:
probably should be able to configure the port that etcd listens on to avoid conflicts...